### PR TITLE
Infra readme

### PR DESCRIPTION
**Pull request by glerzing**
In the context of the #372 (Nginx compression), I made a separate PR for this. Personally I have had much fewer problems by renting a cloud server than by using a VM (problems with ansible, and creating/configuring the VM). And it is much more comfortable, since my computer is not slowed and I don't have to remake all the ansible process after a restart.
So I wanted to indicate this (maybe obvious) alternative for others. Tell me if you think it is an improvement. 
Also, there was a missing dependency for rsync.

### CODE REVIEW

**Comment by glerzing** [comment]
(maybe in this case "tournesol-test" or "tournesol-dev" would be better than "tournesol-vm")

File: infra/README.md
Comment (glerzing) [inline infra/README.md:7]: je n'ai pas précisé mais vu que c'est du HTTP, tout est en clair et ça peut peut-être poser des problèmes de confidentialité

Comment (jstainer-kleis) [reply infra/README.md:7]: with a public IP you can have HTTPS with Let's Encrypt certificates, it only requires setting an email address in the inventory (as specified in a comment)

Comment (jstainer-kleis) [reply infra/README.md:7]: > Renting a server in the cloud is a simpler and faster way to get started
it's really subjective, I personally think it's the contrary

**Comment by jstainer-kleis** [comment]
I'm not opposed to the addition, not really convinced by the value either, I think it's an obvious alternative. I'll let @amatissart decide whether or not to accept it.

File: infra/README.md
Comment (jstainer-kleis) [inline infra/README.md:4]: dev-env is a directory not a respository

Comment (jstainer-kleis) [inline infra/README.md:4]: "less practical", it depends for what. It's meant to test infrastructure modifications and software modification in a prod-like environment. It's less practical while developing because you need to redeploy instead of having hot reload.

File: infra/README.md
Comment (jstainer-kleis) [inline infra/README.md:44]: I'm not sure why the graphical interface would even be installed in the first place.
Even if it's the case, `systemctl disable gdm` should do the job, I'm not sure of the impact of changing the default boot target.

**Review by jstainer-kleis — CHANGES_REQUESTED** [review]

**Comment by glerzing** [comment]
> I'm not opposed to the addition, not really convinced by the value either, I think it's an obvious alternative. I'll let @amatissart decide whether or not to accept it.

I understand that it feels obvious when you have some experience with web deployment. If you think this won't help newcomers to get started more quickly, I can remove this PR.

File: infra/README.md
Comment (glerzing) [reply infra/README.md:4]: indeed I meant directory

File: infra/README.md
Comment (glerzing) [reply infra/README.md:44]: "software selection: only SSH server and standard system utilities" => I probably didn't uncheck something during the installation. I will remove the line I added.

**Comment by GresilleSiffle** [comment]
> I'm not opposed to the addition, not really convinced by the value either, I think it's an obvious alternative. I'll let @amatissart decide whether or not to accept it.

I have the same feeling, but I think we can still get benefit from an improvement of the infrastructure documentation. If @glerzing had encountered issues during the environment set up, it may mean our documentation is not adapted.

I propose to have a discussion next week @glerzing , with @amatissart ( and @jstainer-kleis if you want ) so that you can explain us what exactly are the issues you faced, and how we can improve the `README.md`. Having used a remote virtual machine might mean our instructions regarding the deployment of a local one are incomplete.

What do you think @glerzing ?

**Comment by jstainer-kleis** [comment]
> > I'm not opposed to the addition, not really convinced by the value either, I think it's an obvious alternative. I'll let @amatissart decide whether or not to accept it.
> 
> I have the same feeling, but I think we can still get benefit from an improvement of the infrastructure documentation. If @glerzing had encountered issues during the environment set up, it may mean our documentation is not adapted.
> 
> I propose to have a discussion next week @glerzing , with @amatissart ( and @jstainer-kleis if you want ) so that you can explain us what exactly are the issues you faced, and how we can improve the `README.md`. Having used a remote virtual machine might mean our instructions regarding the deployment of a local one are incomplete.
> 
> What do you think @glerzing ?

Fine with me, I agree that documentation of the infrastructure code may be improved and that people experiencing difficulties are in a good position to pinpoint the tricky parts.
My concerns revolve around the fact that the details of how to create the base VM vary a lot from one context to another, locally as well as remotely. The current documentation was written under the assumption that the user would be able to do this step in her own context.
Notify me if you organize a meeting, I'll try to join (but feel free to do without me in any case).

**Comment by amatissart** [comment]
Thanks @glerzing for looking into this.

I agree with @jstainer-kleis that instantiating a remote VM is not necessarily simpler for newcomers. So it's probably not something we want to support or encourage as the "Option 1", especially as the configuration details may vary a lot depending on the provider.

It's nice to have a reminder here that "dev-env" should be preferred by contributors to test their changes. Perhaps the instructions about using a "cloud server" could be changed into a side note for people who know what they're doing?

**Comment by glerzing** [comment]
I discussed with JST and agreed about the changes. I also set "renting a cloud server" as the option 2 following Adrien's comment.

**Comment by glerzing** [comment]
The failed end-to-end test has nothing to do with this PR. I can reproduce this on my environment since I rebased. For some reason, the frontend container installation is just too slow and the script doesn't wait long enough for the frontend container (=> "frontend is unreachable")

File: infra/README.md
Comment (GresilleSiffle) [inline infra/README.md:9]: I suggest to remove this whole part also. The extra information contained in the " Renting " a server paragraph can be move below. See my next suggestions.

```suggestion
```

File: infra/README.md
Comment (GresilleSiffle) [inline infra/README.md:12]: Title already defined in a previous suggestion.

```suggestion
```

**Review by GresilleSiffle — CHANGES_REQUESTED** [review]
Thanks @glerzing for highlighting the issues you've had with a local virtual machine installation, and for having proposed an alternative.

I didn't have enough time last week to schedule a meeting but after having discussed with @amatissart , here are my suggestions.

My objectives were to preserve the fact that there is an alternative to the local VM setup, and to explicitly notice the users that this method is no officialy supported (even if it should work).

I tried to factorize the installation instructions to avoid duplications, and I deliberately removed the mention of the Cloud providers name to not make any advertisement for them. I don't know them all, and I think recommending Cloud providers is an importact act as they have different policy regarding the data privacy.

**Comment by glerzing** [comment]
I'm ok with GresilleSiffle's suggestions, it makes the Readme lighter.

**Review by GresilleSiffle — APPROVED** [review]

**Comment by GresilleSiffle** [comment]
approved :+1: 

The PR is still waiting for the @jstainer-kleis approval.

**Pull request merged**
